### ISSUE TITLE
`config["snapshots"]["segmentation"]` option breaks sector-coupled networks

### ISSUE THREAD

**Issue by koen-vg** [PyPSA/pypsa-eur#897]
## Checklist

- [x] I am using the current [`master`](https://github.com/PyPSA/pypsa-eur/tree/master) branch or the latest [release](https://github.com/PyPSA/pypsa-eur/releases). Please indicate.
- [x] I am running on an up-to-date [`pypsa-eur` environment](https://github.com/PyPSA/pypsa-eur/blob/master/envs/environment.yaml). Update via `conda env update -f envs/environment.yaml`.

## Describe the Bug

I was delighted that time aggregation and segmentation can now be configured in the `config["snapshots"]` section for those times that you want time resolution to be the same across all model runs, and it's redundant to have it as a constant option in a wildcard.

_However_, this behaviour currently breaks sector-coupled networks. There are two places in the code where time aggregation / segmentation can happen:
1. In `prepare_network.py` (via the `{opts}` wildcard),
2. In `prepare_sector_network.py` (via the `{sector_opts}` wildcard).
The problem is that time aggregation specified in `config["snapshots"]` is applied in `prepare_network.py` regardless of whether any `{sector_opts}` are present, i.e. also for sector-coupled networks. In that case, the network is time-aggregated _before_ being processed in `prepare_sector_network.py`.

In my case, I have at least seen that heat demand is wrongly calculated (much to large, in the order of 100000TWh/a) when aggregating to 50 segments using the `config["snapshots"]` methods. Since `prepare_sector_network.py` is simply not written with time-aggregated networks in mind (I guess), it's entirely possible that there are other problems too.

I have only tried segmentation, but have to assume that plain aggregation also leads to problems.

I see a few options, which is why I'm opening an issue and not just a PR:
1. Very clearly give notice in the documentation and default config that the time aggregation config is _only_ to be used for electricity-only networks;
2. Add some logic in `prepare_network.py` to not do any time aggregation if the `{sector_opts}` wildcard is defined, and add some logic in `prepare_sector_network.py` to fall back to the time aggregation config in a similar way to how it's current done in `prepare_network.py`.
3. Make `prepare_sector_network.py` compatible with time-aggregated input networks. This would involve fixing the heat demand calculation at the very least; possibly more. Implementation could be verified by comparing networks having been time-aggregated before and after `prepare_sector_network.py`. Certainly the "cleanest" option (would also eliminate duplicated time aggregation code between `prepare_network.py` and `prepare_sector_network.py`. Potential drawback is that at least time segmentation has "more information" (especially heat, industrial load) to base the segmentation on _after_ the processing in `prepare_sector_network.py` so the segmentation would potentially be of better quality. However my conjecture would be that this isn't a significant problem in practice because time segmentation should mostly be based on weather variables anyway and the sector coupling doesn't add much significant time-varying data.

The first two options are close to trivial and I would happily open a PR for those. The third option has the potential to be a bit more involved and I don't have time right at the moment to implement it. Could also do e.g. options 2 quickly for now and leave option 3 for a later time.

I do think it would be good to do _something_ about this relatively quickly because a rather natural use-case (time aggregation in config) currently leads to very wrong results (_not_ a workflow error, i.e. more difficult to spot!) for reasons that are probably not easy to track down for most users.

**Comment by fneum** [comment]
Yes, I will look at it today! Quite a severe bug. Tending towards separating the configuration options for clustering in `prepare_network.py` and `prepare_sector_network.py`.
